src/server/responder/serving.cpp

```diff
@@ -86,6 +88,20 @@ void HTTPResponder::serveFile(HTTPClient &client) {
 	// if "*" set to first language in config
 	if (client.data.request.data.acceptLanguage == "*")
 		client.data.request.data.acceptLanguage = client.routeBlock->getAcceptLanguage()[0];
+	
+	// loops through plugins and executes if plugin is loaded
+	for (plugin::PluginContainer::pluginIterator it = globalPlugins.begin(); it != globalPlugins.end(); ++it) {
+		try {
+			if (it->second && it->first->onBeforeFileServing(client)) {
+				client.data.response.setResponse(client.data.response.builder.build());
+				return;
+			}
+		}
+		catch (const std::exception& e) {
+			globalLogger.logItem(logger::ERROR, std::string("Json_stat_API error: ") + e.what());
+			handleError(client, 404);	
+		}
+	}
```

Comment:
Move the hook to line 35 of this file. that will also include serving of index files